### ISSUE TITLE
Split descriptions are of insufficient precision

### ISSUE THREAD

**Issue by BenModulusLabs** [catboost/catboost#2415]
Problem: Split descriptions are of insufficient precision (see below)
catboost version: 1.2
Operating System: Mac OS X 13.3.1
CPU: Apple Silicon M2
GPU: N/A

The splits used by Catboost are single precision floats, but the split descriptions available via the Python interface use only 6 significant digits.  This level of precision is insufficient to fully specify the 32-bit float (8 significant digits are required).  The result is that distinct but very close together split thresholds appear to identical, and the split information can not be used to trace paths through the decision tree.

(The workaround I've found is to export the model to JSON, and fetch the splits from there, but this seems not ideal).

Example:
```python
model = CatBoostRegressor().load_model('mymodel.h5')
tree_idx = 0
splits = model._get_tree_splits(tree_idx, pool=None)
splits[:5]
```

Outputs:
```
['27, bin=0.914953',
 '32, bin=-0.76942',
 '6, bin=-2.20777',
 '32, bin=-2.43341',
 '13, bin=-3.83913']
```

**Comment by mihail-shusharin** [comment]
Hello, function _get_tree_splits is used only for tree visualisation and should't return 8 significant digits. Your workaround seems to be the best way to fetch splits to use them somewhere else.

**Comment by BenModulusLabs** [comment]
Yes, I agree in principle, but as it currently stands the JSON export is insufficient to reconstruct the tree because of this issue https://github.com/catboost/catboost/issues/2414. Presently it is necessary to use both `_get_tree_splits` and the JSON export together!

It would moreover be more convenient to be able to retrieve the tree information from the Catboost model object directly, without having to write the JSON to disk and read it back in again.  So perhaps the solution is to create a new public method in the Python API that returns the split information (and ideally not as a string)?
